spikeinterface/preprocessing/highpass_spatial_filter.py

```diff
@@ -0,0 +1,290 @@
+        is applied prior to filtering to improve filter performance.
+        When the argument is "ibl", the function will use the IBL pipeline defaults (agc on).
+        Setting to None will turn off gain control, by default None.
+        To customize, pass a dictionary with the fields:
+            * agc_options["window_length_s"] - window length in seconds
+            * agc_options["sampling_interval"] - recording sampling interval
+    butter_kwargs: dict
+        Dictionary with fields "N" and "Wn" to be passed to
+        scipy.signal.butter, by default N=3 and Wn=0.01
+    """
+    name = 'highpass_spatial_filter'
+
+    def __init__(self, recording, n_channel_pad=None, n_channel_taper=5, direction="y",
+                 agc_options=None, butter_kwargs=None):
+        BasePreprocessor.__init__(self, recording)
+
+        # Check single group
+        channel_groups = recording.get_channel_groups()
+        assert len(np.unique(channel_groups)) == 1, \
+            ("The recording contains multiple groups! It is recommended to apply spatial filtering "
+             "separately for each group. You can split by group with: "
+             ">>> recording_split = recording.split_by(property='group')")
+        # If location are not sorted, estimate forward and reverse sorting
+        channel_locations = recording.get_channel_locations()
+        dim = ["x", "y", "z"].index(direction)
+        assert dim < channel_locations.ndim, f"Direction {direction} is wrong"
+        locs_mono = channel_locations[:, dim]
+        if np.array_equal(np.sort(locs_mono), locs_mono):
+            order_f = None
+            order_r = None
+        else:
+            order_f = np.argsort(locs_mono)
+            order_r = np.argsort(order_f)
+
+        # Fix channel padding and tapering
+        n_channel_pad = 0 if n_channel_pad is None else int(n_channel_pad)
+        assert n_channel_pad <= recording.get_num_channels(), \
+            "'n_channel_pad' must be less than the number of channels in recording."
+        n_channel_taper = n_channel_pad if n_channel_taper is None else int(n_channel_taper)
+        assert n_channel_taper <= recording.get_num_channels(), \
+            "'n_channel_taper' must be less than the number of channels in recording."
+
+        # Fix Automatic Gain Control options
+        if agc_options is not None:
+            assert isinstance(agc_options, (str, dict)), \
+                f"agc_options can be 'ibl', a dictionary, or None, not {type(agc_options)}"
+            sampling_interval = 1 / recording.sampling_frequency
+            if isinstance(agc_options, str):
+                assert agc_options == "ibl", "agc_options can be 'ibl', a dictionary, or None"
+                # default IBL value is 300 * sampling_interval
+                default_window_length = 300 * sampling_interval 
+                agc_options = {"window_length_s": default_window_length,
```

Comment:
Here this is strange `window_length_s` make think that it is in second but this is in sample in fact no ?

Reply:
Oups no sorry I missread. this is OK.